### PR TITLE
fix: bump PATCH dependencies

### PR DESCRIPTION
**Pull request by jtmalinowski**
# Description

Bump development dependencies by `minor`, and OTel by `patch`.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] Tested manually
- [x] Added automated tests

# Checklist:

- [ ] Changelogs have been updated
- [x] Unit tests have been added/updated
- [ ] Documentation has been updated

### CODE REVIEW

**Review by Rauno56 — APPROVED** [review]

File: test/propagation.test.ts [hunk 1]
```diff
@@ -24,7 +24,11 @@ import {
 } from '@opentelemetry/api';
 import { startTracing, stopTracing } from '../src/tracing';
 import { CompositePropagator, RandomIdGenerator } from '@opentelemetry/core';
-import { InMemorySpanExporter, SpanProcessor } from '@opentelemetry/tracing';
+import {
+  InMemorySpanExporter,
+  ReadableSpan,
```
Comment (Rauno56) [inline test/propagation.test.ts:29]: That's here from some experimentation, I guess, and the fact that nothing notifies us tells me we don't lint in CI.

**Review by Rauno56 — APPROVED** [review]
See the comment. Otherwise, mergeable!

File: test/propagation.test.ts (hunk 1 above)
Comment (jtmalinowski) [reply test/propagation.test.ts:29]: We do, I think this is `prettier` in action.

Comment (Rauno56) [reply test/propagation.test.ts:29]: prettier def doesn't add imports.

**Pull request merged**